projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -1041,6 +1041,12 @@ function ( $domain ) {
  * @return array|WP_Error Array of domains and their verification status or WP_Error if the request fails.
  */
 function wpcom_request_domains_list() {
+	static $cached_domains = null;
+
+	if ( $cached_domains !== null ) {
+		return $cached_domains;
```

Comment:
This was a suggestion by Dale, just in case we call this function more than once in the same request in the future, preventing we request to WPCOM from Atomic more than once.

Reply:
I think it's worth calling out more details about the caching mechanism and how it's supposed to work in the code and in the PR. For the inline comments, I'd phrase this as follows:
```php
	// Use a static variable as a temporary in-memory cache to avoid multiple outbound 
	// HTTP requests within a single incoming request.
	// We don't expect this to be triggered multiple times, but it's worth adding some 
	// light caching to avoid multiple, possibly slow HTTP requests where the underlying data 
	// is highly unlikely to change.
	// The "cache" only lasts as long as the current request/memory space, so we don't need to invalidate it.
	static $cached_domains = null;
```

While writing the above, I realised there is one case where we may need to be careful: shell/CLI sessions where we load the code once and keep the same memory space. We may want to skip the caching when we're in that context (especially WP CLI and our internal wpsh), but that may be something for a follow-up PR rather than delaying this one.